packages/services/src/view-service/fees.ts

```diff
@@ -1,15 +1,27 @@
 import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
 import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import { IndexedDbInterface } from '@penumbra-zone/types/indexed-db';
 
-// Attempts to extract a fee token from the assets used in the actions of the planner request
-// Priority in descending order
-export const extractAltFee = (request: TransactionPlannerRequest): AssetId => {
+export const extractAltFee = async (
+  request: TransactionPlannerRequest,
+  indexedDb: IndexedDbInterface,
+): Promise<AssetId> => {
   const outputAsset = request.outputs.map(o => o.value?.assetId).find(Boolean);
   if (outputAsset) return outputAsset;
 
   const swapAsset = request.swaps.map(assetIn => assetIn.value?.assetId).find(Boolean);
   if (swapAsset) return swapAsset;
 
+  const swapCommitment = request.swapClaims
+    .map(swapClaim => swapClaim.swapCommitment)
+    .find(commitment => commitment !== undefined);
+  if (swapCommitment) {
+    const swaps = await indexedDb.getSwapByCommitment(swapCommitment);
+    if (swaps?.swap?.tradingPair?.asset1) {
+      return swaps.swap.tradingPair.asset1;
+    }
+  }
```

Comment:
I believe this is saying "If a user is making a swap claim, attempt to find the assetIn they originally swapped with for an alt fee".

But given swap claims happen in different blocks than swaps, isn't it possible the user doesn't have that original asset anymore? Like what if they swapped the entire balance of it out? Is this logic we can't really depend on?

Reply:
as discussed, the original asset value balances are irrelevant since the swap pre-pays the swap claim. The assetId however must be consistent between the swap / swap claim and can be retrieved by querying the swap table in the database or by calling the `SwapByCommitment` service implementation using the swap commitment as the key.